<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="David Haberthür" />
  <meta name="author" content="Oleksiy-Zakhar Khoma" />
  <meta name="author" content="Tim Hoessly" />
  <meta name="author" content="Eugenio Zoni" />
  <meta name="author" content="Marianna Kruithof-de Julio" />
  <meta name="author" content="Stewart D. Ryan" />
  <meta name="author" content="Myriam Grunewald" />
  <meta name="author" content="Benjamin Bellón" />
  <meta name="author" content="Rebecca Sandgren" />
  <meta name="author" content="Benjamin E. Pippenger" />
  <meta name="author" content="Dieter Bosshardt" />
  <meta name="author" content="Valentin Djonov" />
  <meta name="author" content="Ruslan Hlushchuk" />
  <meta name="dcterms.date" content="2023-06-01" />
  <meta name="keywords" content="microCT, contrast agent, microvasculature" />
  <title>MicroCT-based imaging of microvasculature within the bone tissue</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta property="og:type" content="article" />
  <meta name="dc.title" content="MicroCT-based imaging of microvasculature within the bone tissue" />
  <meta name="citation_title" content="MicroCT-based imaging of microvasculature within the bone tissue" />
  <meta property="og:title" content="MicroCT-based imaging of microvasculature within the bone tissue" />
  <meta property="twitter:title" content="MicroCT-based imaging of microvasculature within the bone tissue" />
  <meta name="dc.date" content="2023-06-01" />
  <meta name="citation_publication_date" content="2023-06-01" />
  <meta property="article:published_time" content="2023-06-01" />
  <meta name="dc.modified" content="2023-06-01T10:12:59+00:00" />
  <meta property="article:modified_time" content="2023-06-01T10:12:59+00:00" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="David Haberthür" />
  <meta name="citation_author_institution" content="Institute of Anatomy, University of Bern, Baltzerstrasse 2, CH-3012 Bern, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0003-3388-9187" />
  <meta name="citation_author" content="Oleksiy-Zakhar Khoma" />
  <meta name="citation_author_institution" content="Institute of Anatomy, University of Bern, Baltzerstrasse 2, CH-3012 Bern, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-1914-6873" />
  <meta name="citation_author" content="Tim Hoessly" />
  <meta name="citation_author_institution" content="Institute of Anatomy, University of Bern, Baltzerstrasse 2, CH-3012 Bern, Switzerland" />
  <meta name="citation_author_orcid" content="None" />
  <meta name="citation_author" content="Eugenio Zoni" />
  <meta name="citation_author_institution" content="Department for BioMedical Research, Urology Research Laboratory, University of Bern, Bern, Switzerland" />
  <meta name="citation_author_institution" content="Department of Urology, Inselspital, Bern University Hospital, Bern, Switzerland" />
  <meta name="citation_author_orcid" content="None" />
  <meta name="citation_author" content="Marianna Kruithof-de Julio" />
  <meta name="citation_author_institution" content="Department for BioMedical Research, Urology Research Laboratory, University of Bern, Bern, Switzerland" />
  <meta name="citation_author_institution" content="Department of Urology, Inselspital, Bern University Hospital, Bern, Switzerland" />
  <meta name="citation_author_institution" content="Department for BioMedical Research, Translation Organoid Research, University of Bern, Bern, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-6085-7706" />
  <meta name="citation_author" content="Stewart D. Ryan" />
  <meta name="citation_author_institution" content="U-Vet Hospital, University of Melbourne Veterinary School, Werribee, Australia" />
  <meta name="citation_author_orcid" content="0000-0001-7016-771X" />
  <meta name="citation_author" content="Myriam Grunewald" />
  <meta name="citation_author_institution" content="Department of Developmental Biology and Cancer Research, Faculty of Medicine, The Hebrew University, Organoid Research Center, Hadassah University Hospital, Jerusalem, Israel" />
  <meta name="citation_author_orcid" content="0000-0001-5394-9218" />
  <meta name="citation_author" content="Benjamin Bellón" />
  <meta name="citation_author_institution" content="Preclinical and Translational Research, Institut Straumann AG, Basel, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0003-2297-9160" />
  <meta name="citation_author" content="Rebecca Sandgren" />
  <meta name="citation_author_institution" content="Centre for Comparative Medicine, Medical Faculty, Lund University, Lund, Sweden" />
  <meta name="citation_author_orcid" content="0000-0003-4473-4666" />
  <meta name="citation_author" content="Benjamin E. Pippenger" />
  <meta name="citation_author_institution" content="Preclinical and Translational Research, Institut Straumann AG, Basel, Switzerland" />
  <meta name="citation_author_institution" content="Department of Periodontology, University of Bern, Bern, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0001-7990-5555" />
  <meta name="citation_author" content="Dieter Bosshardt" />
  <meta name="citation_author_institution" content="Department of Periodontology, University of Bern, Bern, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-2132-6363" />
  <meta name="citation_author" content="Valentin Djonov" />
  <meta name="citation_author_institution" content="Institute of Anatomy, University of Bern, Baltzerstrasse 2, CH-3012 Bern, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-5062-1169" />
  <meta name="citation_author" content="Ruslan Hlushchuk" />
  <meta name="citation_author_institution" content="Institute of Anatomy, University of Bern, Baltzerstrasse 2, CH-3012 Bern, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-6722-8996" />
  <link rel="canonical" href="https://microct-ana-unibe-ch.github.io/microvasculature-manuscript/" />
  <meta property="og:url" content="https://microct-ana-unibe-ch.github.io/microvasculature-manuscript/" />
  <meta property="twitter:url" content="https://microct-ana-unibe-ch.github.io/microvasculature-manuscript/" />
  <meta name="citation_fulltext_html_url" content="https://microct-ana-unibe-ch.github.io/microvasculature-manuscript/" />
  <meta name="citation_pdf_url" content="https://microct-ana-unibe-ch.github.io/microvasculature-manuscript/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://microct-ana-unibe-ch.github.io/microvasculature-manuscript/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://microct-ana-unibe-ch.github.io/microvasculature-manuscript/v/90cc2c40ff05945e53f3d30945f3915aefa88290/" />
  <meta name="manubot_html_url_versioned" content="https://microct-ana-unibe-ch.github.io/microvasculature-manuscript/v/90cc2c40ff05945e53f3d30945f3915aefa88290/" />
  <meta name="manubot_pdf_url_versioned" content="https://microct-ana-unibe-ch.github.io/microvasculature-manuscript/v/90cc2c40ff05945e53f3d30945f3915aefa88290/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
</head>
<body>
<header id="title-block-header">
<h1 class="title">MicroCT-based imaging of microvasculature within the bone tissue</h1>
</header>
<p><em>A preprint of this manuscript is available at bioRχiv with the <a href="https://doi.org/10.1101/2023.03.08.531678">doi:10.1101/2023.03.08.531678</a>.</em>
<em>We submitted <a href="https://github.com/microct-ana-unibe-ch/microvasculature-manuscript/releases/tag/v1.0"><code>v1.0</code></a> of the manuscript to <a href="http://www.small-methods.com/">Small Methods</a>.</em>
<em>The version you see here <a href="https://github.com/microct-ana-unibe-ch/microvasculature-manuscript/compare/v1.0...90cc2c40ff05945e53f3d30945f3915aefa88290#files_bucket">has been updated</a> since submission.</em></p>
<p><small><em>
This manuscript
(<a href="https://microct-ana-unibe-ch.github.io/microvasculature-manuscript/v/90cc2c40ff05945e53f3d30945f3915aefa88290/">permalink</a>)
was automatically generated
from <a href="https://github.com/microct-ana-unibe-ch/microvasculature-manuscript/tree/90cc2c40ff05945e53f3d30945f3915aefa88290">microct-ana-unibe-ch/microvasculature-manuscript@90cc2c4</a>
on June 1, 2023.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>David Haberthür</strong>
<sup><a href="#equal_contribution">☯</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-3388-9187">0000-0003-3388-9187</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/habi">habi</a>
· <img src="images/mastodon.svg" class="inline_icon" width="16" height="16" alt="Mastodon icon" />
<a href="https://mastodon.social/@habi">@habi@mastodon.social</a>
<br>
<small>
Institute of Anatomy, University of Bern, Baltzerstrasse 2, CH-3012 Bern, Switzerland
</small></p></li>
<li><p><strong>Oleksiy-Zakhar Khoma</strong>
<sup><a href="#equal_contribution">☯</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-1914-6873">0000-0002-1914-6873</a>
· <img src="images/github.svg" class="inline_icon" width="16" height="16" alt="GitHub icon" />
<a href="https://github.com/ol-khoma">ol-khoma</a>
<br>
<small>
Institute of Anatomy, University of Bern, Baltzerstrasse 2, CH-3012 Bern, Switzerland
</small></p></li>
<li><p><strong>Tim Hoessly</strong>
<br>
<small>
Institute of Anatomy, University of Bern, Baltzerstrasse 2, CH-3012 Bern, Switzerland
</small></p></li>
<li><p><strong>Eugenio Zoni</strong>
<br>
<small>
Department for BioMedical Research, Urology Research Laboratory, University of Bern, Bern, Switzerland; Department of Urology, Inselspital, Bern University Hospital, Bern, Switzerland
</small></p></li>
<li><p><strong>Marianna Kruithof-de Julio</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-6085-7706">0000-0002-6085-7706</a>
<br>
<small>
Department for BioMedical Research, Urology Research Laboratory, University of Bern, Bern, Switzerland; Department of Urology, Inselspital, Bern University Hospital, Bern, Switzerland; Department for BioMedical Research, Translation Organoid Research, University of Bern, Bern, Switzerland
</small></p></li>
<li><p><strong>Stewart D. Ryan</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7016-771X">0000-0001-7016-771X</a>
<br>
<small>
U-Vet Hospital, University of Melbourne Veterinary School, Werribee, Australia
</small></p></li>
<li><p><strong>Myriam Grunewald</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-5394-9218">0000-0001-5394-9218</a>
<br>
<small>
Department of Developmental Biology and Cancer Research, Faculty of Medicine, The Hebrew University, Organoid Research Center, Hadassah University Hospital, Jerusalem, Israel
</small></p></li>
<li><p><strong>Benjamin Bellón</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-2297-9160">0000-0003-2297-9160</a>
<br>
<small>
Preclinical and Translational Research, Institut Straumann AG, Basel, Switzerland
</small></p></li>
<li><p><strong>Rebecca Sandgren</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-4473-4666">0000-0003-4473-4666</a>
<br>
<small>
Centre for Comparative Medicine, Medical Faculty, Lund University, Lund, Sweden
</small></p></li>
<li><p><strong>Benjamin E. Pippenger</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7990-5555">0000-0001-7990-5555</a>
<br>
<small>
Preclinical and Translational Research, Institut Straumann AG, Basel, Switzerland; Department of Periodontology, University of Bern, Bern, Switzerland
</small></p></li>
<li><p><strong>Dieter Bosshardt</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-2132-6363">0000-0002-2132-6363</a>
<br>
<small>
Department of Periodontology, University of Bern, Bern, Switzerland
</small></p></li>
<li><p><strong>Valentin Djonov</strong>
<br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-5062-1169">0000-0002-5062-1169</a>
<br>
<small>
Institute of Anatomy, University of Bern, Baltzerstrasse 2, CH-3012 Bern, Switzerland
</small></p></li>
<li><p><strong>Ruslan Hlushchuk</strong>
<sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" width="16" height="16" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-6722-8996">0000-0002-6722-8996</a>
<br>
<small>
Institute of Anatomy, University of Bern, Baltzerstrasse 2, CH-3012 Bern, Switzerland
</small></p></li>
</ul>
<div id="equal_contribution">
<p>☯ — David Haberthür and
Oleksiy-Zakhar Khoma contributed equally to this work.</p>
</div>
<div id="correspondence">
<p>✉ — Corresponding author: Ruslan Hlushchuk (<a href="mailto:ruslan.hlushchuk@unibe.ch">ruslan.hlushchuk@unibe.ch</a>, <a href="tel:+41%2031%20684%2046%2080">+41 31 684 46 80</a>).</p>
</div>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>Angiogenesis is essential for skeletal development, bone healing and regeneration.
Improved non-destructive, three-dimensional (3D) imaging of the vasculature within bone tissue would benefit a multitude of research areas, especially implantology and tissue engineering.</p>
<p>X-ray microtomography (microCT) is a well-suited non-destructive 3D imaging technique for bone morphology.
For the detection of vessels, using contrast-enhanced microCT imaging is essential.
Limited contrast between perfusion agents and mineralized bone has been the major drawback of this approach, making their distinct segmentation problematic.
A decalcification step resolves this issue but inhibits simultaneous assessment of intracortical bone microstructure and vascular morphology.
Moreover, the problem of contrasting becomes further compounded in samples with metal implants.</p>
<p>This study describes µAngiofil-enhanced microCT-based visualization of vasculature within bone tissue in small and large animal models, with and without decalcification.
We present simultaneous microvascular and bone imaging in murine tibia, murine bone metastatic model, pulp chamber, gingiva and periodontal ligaments.
In a large animal model (minipig) we perform visualization and segmentation of different tissue types and vessels in the hemimandible containing metal implants.</p>
<p>Our manuscript introduces the first non-destructive approach for 3D imaging of the vasculature within soft and hard tissues in the vicinity of metal implants in a large animal model.</p>
<h2 class="page_break_before" id="introduction">Introduction</h2>
<p>Angiogenesis is an essential physiological vascularization process for skeletal development and growth as well as for bone healing and regeneration.
This formation of new blood vessels from pre-existing vessels is crucial for successful bone formation and repair <span class="citation" data-cites="zXoIXG3G AHsHFIUt 1FaKEBtU8 Vu36g0cK">[<a href="#ref-yZTJBX1B" role="doc-biblioref">5</a>]</span>.
In addition to carrying nutrients and growth factors, those newly formed blood vessels are a delivery route of stem cells and progenitor cells to the bone-defect site <span class="citation" data-cites="dBkW39I8 Es64ZQO 5mgeUNd6">[<a href="#ref-dBkW39I8" role="doc-biblioref">6</a>,<a href="#ref-Es64ZQO" role="doc-biblioref">7</a>,<a href="#ref-5mgeUNd6" role="doc-biblioref">8</a>]</span>.
The structural nature of skeletal tissue makes three-dimensional (3D) imaging of its vasculature extremely difficult.
Histology, a destructive and two-dimensional approach, still remains a gold standard for assessing vasculature in bones <span class="citation" data-cites="117R9ULnv">[<a href="#ref-117R9ULnv" role="doc-biblioref">9</a>]</span>.
Classic soft tissue imaging techniques such as light sheet microscopy or confocal laser scanning microscopy face challenges in their application due to the encasement of blood vessels in calcified tissue <span class="citation" data-cites="117R9ULnv">[<a href="#ref-117R9ULnv" role="doc-biblioref">9</a>]</span>.
Despite recent advancements in tissue clearing-based imaging methods for craniofacial bones in a mouse model <span class="citation" data-cites="DiX3YTwQ X0aG1K2m">[<a href="#ref-DiX3YTwQ" role="doc-biblioref">10</a>,<a href="#ref-X0aG1K2m" role="doc-biblioref">11</a>]</span>, the application of such methods remains challenging.
In the case of bone grafts, many of the synthetic ones fail to bridge critically sized defects due to their inability to promote vascularization <span class="citation" data-cites="33Vq8bQx Vu36g0cK">[<a href="#ref-Vu36g0cK" role="doc-biblioref">4</a>,<a href="#ref-33Vq8bQx" role="doc-biblioref">12</a>]</span>.
In essence, simultaneous non-destructive 3D imaging of the vasculature within bone tissue and of the bone tissue itself, especially in the case of larger bone grafts, has been a challenge for decades <span class="citation" data-cites="1FUMXJVtA Vu36g0cK">[<a href="#ref-Vu36g0cK" role="doc-biblioref">4</a>,<a href="#ref-1FUMXJVtA" role="doc-biblioref">13</a>]</span>.
Thus, many research areas benefit from an improved three-dimensional (3D) imaging of the vasculature within bone tissue: for example bone metastatic disease, bone biology, tissue engineering, implantology, reconstructive surgery and healing of both small and critical size bone defects with or without bone grafts.</p>
<p>In the last decades, X-ray micro-computed tomography (microCT) gained recognition as a non-destructive 3D imaging technique for bone morphology <span class="citation" data-cites="KjvwGWK2">[<a href="#ref-KjvwGWK2" role="doc-biblioref">14</a>]</span>.
Due to the inherently low difference in X-ray absorption levels between vessels and different soft tissues it is not easily feasible to distinguish such structures within the bone.
To unambiguously detect vasculature within bone it is thus necessary to instill the vessels with either a contrast agent or use a casting method to otherwise fill the blood vessels.
Currently existing protocols for imaging a the vasculature within the bone via a vascular replica, have drawbacks like showing disjoint vascular components or completely missing vascular segments <span class="citation" data-cites="168Ee16HU 117R9ULnv">[<a href="#ref-117R9ULnv" role="doc-biblioref">9</a>,<a href="#ref-168Ee16HU" role="doc-biblioref">15</a>]</span>.
It was also reported that the contrast difference between the perfusion agent to generate the replica of the vascular network an the mineralized bone make it problematic to perform a distinct segmentation of the bone tissue and the vasculature.
This issue can be overcome by decacifying samples prior to vascular imaging.
Such a decalcification procedure makes simultaneous assessment of the intracortical bone microstructure and the vascular morphology impossible, though <span class="citation" data-cites="117R9ULnv">[<a href="#ref-117R9ULnv" role="doc-biblioref">9</a>]</span>.</p>
<p>Intravascular contrast-agent-enhanced microCT has the potential to overcome these issues.
It has become a method of choice for the evaluation of angiogenesis in bone tissue engineering and remodeling applications <span class="citation" data-cites="13YRlKdu2 AHsHFIUt Vu36g0cK">[<a href="#ref-AHsHFIUt" role="doc-biblioref">2</a>,<a href="#ref-Vu36g0cK" role="doc-biblioref">4</a>,<a href="#ref-13YRlKdu2" role="doc-biblioref">16</a>]</span>.
Barium sulfate and Microfil have been applied as the two most common contrast agents in studies on the vascularization of bone tissue.
Only some selected studies <span class="citation" data-cites="AHsHFIUt">[<a href="#ref-AHsHFIUt" role="doc-biblioref">2</a>]</span> managed to show better perfustion and thus visualization when perfusing the vasculature with barium sulfate.
Previous studies have reported disadvantages associated with barium sulfate suspensions, including higher viscosity, which can sometimes result in incomplete vascular filling and weak or inhomogeneous signal, particularly in higher resolution scans <span class="citation" data-cites="AHsHFIUt">[<a href="#ref-AHsHFIUt" role="doc-biblioref">2</a>]</span>.
These issues may be attributed to particle aggregation, as indicated by various studies <span class="citation" data-cites="DWiXWNoA KJUjdwi7 Ror9Uhfa">[<a href="#ref-DWiXWNoA" role="doc-biblioref">17</a>,<a href="#ref-KJUjdwi7" role="doc-biblioref">18</a>,<a href="#ref-Ror9Uhfa" role="doc-biblioref">19</a>]</span>.</p>
<p>Although Microfil has probably been applied in a larger number of vasculature imaging studies than barium sulfate, Microfil was reported to have disadvantages like poor or incomplete filling of the vasculature <span class="citation" data-cites="KJUjdwi7 AHsHFIUt Ror9Uhfa">[<a href="#ref-AHsHFIUt" role="doc-biblioref">2</a>,<a href="#ref-KJUjdwi7" role="doc-biblioref">18</a>,<a href="#ref-Ror9Uhfa" role="doc-biblioref">19</a>]</span> or vascular damage <span class="citation" data-cites="1A0HsmRho">[<a href="#ref-1A0HsmRho" role="doc-biblioref">20</a>]</span>.</p>
<p>The problem of contrasting the vasculature becomes even further compounded in the presence of a metal implant.
Such implants have revolutionized the treatment of patients with missing teeth or injured joints and bones <span class="citation" data-cites="11349lK8v rCXuqa0O">[<a href="#ref-11349lK8v" role="doc-biblioref">21</a>,<a href="#ref-rCXuqa0O" role="doc-biblioref">22</a>]</span>.
Interactions occurring at the implant-tissue interface are widely believed to play a crucial role in the success of implant placement and healing <span class="citation" data-cites="11349lK8v 137VU0bph">[<a href="#ref-11349lK8v" role="doc-biblioref">21</a>,<a href="#ref-137VU0bph" role="doc-biblioref">23</a>]</span>.
MicroCT is the only available approach to non-destructively investigate an intact bone-implant interface in 3D <span class="citation" data-cites="11349lK8v">[<a href="#ref-11349lK8v" role="doc-biblioref">21</a>]</span>.</p>
<p>Beyond studying the anatomy and physiology of angiogenesis and vasculature of the bone itself, the contribution of the vascular component in the healing process around bone-borne implants is also considered an area of study meriting further experimentation <span class="citation" data-cites="Vu36g0cK 5CxAuEpv">[<a href="#ref-Vu36g0cK" role="doc-biblioref">4</a>,<a href="#ref-5CxAuEpv" role="doc-biblioref">24</a>]</span>.
To date, only few studies tried to assess and visualize the vasculature surrounding implants using microCT imaging, for examples see <span class="citation" data-cites="uAjlWLJe">[<a href="#ref-uAjlWLJe" role="doc-biblioref">25</a>]</span>, where Wang et al. assessed a region of 100 μm around the implant surface and <span class="citation" data-cites="jhdsbJV3">[<a href="#ref-jhdsbJV3" role="doc-biblioref">26</a>]</span>, where Bhargava et al. present VascuViz, a pipeline for vascular systems biology and compare it to other vascular imaging workflows.
Both these studies image the samples at approximately 7 μm voxel size, inhibiting the assessment of microvessels with a diameter of approximately 40 µm or less.
To the best of our knowledge, no study has managed to resolve the microvascular component in a large animal model due to the technical limitations of the perfused contrast agent and the applied microCT imaging technique.</p>
<p>The present study introduces the polymer-based contrast agent µAngiofil for high-resolution microCT-based visualization of microvasculature within bone tissue in small and large animal models, both with and without decalcification of the bone.
Moreover, we show that µAngiofil is suitable for simultaneous imaging and subsequent analysis of peri-implant hard and soft tissues as well as their vascularization in the vicinity of metal implants in a large animal model.</p>
<h2 class="page_break_before" id="materials-methods-and-results">Materials, Methods and Results</h2>
<h3 id="animals">Animals</h3>
<p>In this study we used one transgenic VEGF male 21 months old mouse (see <span class="citation" data-cites="nC0ZZVgL">[<a href="#ref-nC0ZZVgL" role="doc-biblioref">27</a>]</span> for more details), five 6 weeks old CB17SCID male mice and two Göttingen minipigs.
Animal procedures were performed in accordance with the applicable Swedish, Israeli or Swiss legislation on the protection of animals and were approved by the corresponding committees.</p>
<p>The murine experiment were approved by the local ethical committee of the Canton of Bern, Switzerland under the permit number BE 55/16.
Minipig experiments were approved by the Malmö/Lund regional ethical committee, Sweden under license 5.8.18-15672/2019.</p>
<h3 id="contrast-enhanced-microangioct-of-mice-using-µangiofil">Contrast-enhanced microangioCT of mice using µAngiofil</h3>
<p>The contrast agent µAngiofil (Fumedica AG, Switzerland) was prepared according to the manufacturer’s recommendations.
The perfusion of the mice was performed as previously described <span class="citation" data-cites="s9aC9H1e ijqzbOI6">[<a href="#ref-s9aC9H1e" role="doc-biblioref">28</a>,<a href="#ref-ijqzbOI6" role="doc-biblioref">29</a>]</span>.
Briefly, heparinized animals were deeply anesthetized, the thorax and the peritoneal cavity were opened with scissors to expose the descending aorta.
The exposed aorta was then cannulated in either antegrade (for the perfusion of the hind limbs) or retrograde direction (for the perfusion of the head and teeth) with a Venflon cannula (26 GA).
Afterwards, the blood was flushed out with warm PBS solution.
Several cuts of the liver edge allowed for effusion of the blood and buffer solution.
Thereafter, the selected part of the body (hind limbs or head and teeth) was perfused with µAngiofil at 1-1.5 ml/min using a syringe pump.
The perfusion lasted until the organ of interest appeared completely blue <span class="citation" data-cites="TYAmre95 ijqzbOI6">[<a href="#ref-ijqzbOI6" role="doc-biblioref">29</a>,<a href="#ref-TYAmre95" role="doc-biblioref">30</a>]</span>.
In bones, it is not possible to visually monitor this color change, thus perfusion of the neighboring soft tissues serves as an indirect marker of sufficient perfusion within the bone.
To achieve correct perfusion of the vessels within the bone, we prolonged the perfusion time by instilling at least 2 ml of extra volume of contrast agent after all the superficial tissues of the extremity or head turned blue..
After harvesting, the samples were fixed by immersion in 4% paraformaldehyde (PFA) solution and then imaged using microCT scanners (Bruker microCT, Kontich, Belgium).</p>
<p>Figure <a href="#fig:1">1</a> shows the bone microstructure and vascularization of a tibia of a 21-month-old VEGF transgenic male mouse.
This visualization approach enabled a simultaneous display of bone, its vascularization, and their segmentation.
As presented in our other study, we were able to quantitatively show that the age-related osteoporosis diminishes with improved vascularization <span class="citation" data-cites="nC0ZZVgL">[<a href="#ref-nC0ZZVgL" role="doc-biblioref">27</a>]</span>.</p>
<div id="fig:1" class="fignos">
<figure>
<img src="images/Fig1_murine%20tibia.png" alt="Figure 1: microangioCT of proximal murine tibia of a 21-month-old VEGF transgenic male mouse. After perfusion with µAngiofil the murine tibia was harvested, fixated in 4% PFA and imaged by microCT. Arrowheads mark microvessels within the tibia. The diameter of the tibia shaft is around 1 mm. On the right side of the image one can distinguish the bigger epiphyseal vessels. The bone tissue appears white at the plane of the virtual section through the microCT-dataset. Scanning parameters (SKYSCAN 1172): acceleration voltage 50 kV, current 200 µA, 2.98 µm voxel size, 360 degrees scan, 0.05° rotation step, frame averaging 3." />
<figcaption aria-hidden="true"><span>Figure 1:</span> <strong>microangioCT of proximal murine tibia of a 21-month-old VEGF transgenic male mouse.</strong>
After perfusion with µAngiofil the murine tibia was harvested, fixated in 4% PFA and imaged by microCT.
Arrowheads mark microvessels within the tibia.
The diameter of the tibia shaft is around 1 mm.
On the right side of the image one can distinguish the bigger epiphyseal vessels.
The bone tissue appears white at the plane of the virtual section through the microCT-dataset.
Scanning parameters (SKYSCAN 1172): acceleration voltage 50 kV, current 200 µA, 2.98 µm voxel size, 360 degrees scan, 0.05° rotation step, frame averaging 3.</figcaption>
</figure>
</div>
<h3 id="decalcification-of-the-µangiofil-perfused-murine-tibia">Decalcification of the µAngiofil-perfused murine tibia</h3>
<p>It is practically the standard in the field to decalcify the bone samples to enable the proper visualization and segmentation of vasculature within bone tissue <span class="citation" data-cites="117R9ULnv AHsHFIUt">[<a href="#ref-AHsHFIUt" role="doc-biblioref">2</a>,<a href="#ref-117R9ULnv" role="doc-biblioref">9</a>]</span>.
We successfully established a bone-decalcifying protocol (for murine hind limb) with 10% Ethylenediaminetetraacetic acid (EDTA) solution <span class="citation" data-cites="iM9i6gkJ">[<a href="#ref-iM9i6gkJ" role="doc-biblioref">31</a>]</span> by adapting previous work <span class="citation" data-cites="168Ee16HU">[<a href="#ref-168Ee16HU" role="doc-biblioref">15</a>]</span>.
Decalcification of bone tissue does not negatively influence the bone structure itself, but reduces the X-ray absorption of bone tissue and thus makes it impossible to visualize simultaneously with the vasculature (Fig. <a href="#fig:2">2</a>).
On the other hand, it enables tomographic imaging of the murine hind limb vasculature with less artifacts around the bone at lower acceleration voltages, which in most instances leads to shorter scanning times.
Furthermore, segmentation of the vasculature in the tomographic datasets becomes easier and a subsequent histological evaluation of the sample is facilitated <span class="citation" data-cites="iM9i6gkJ">[<a href="#ref-iM9i6gkJ" role="doc-biblioref">31</a>]</span>.</p>
<div id="fig:2" class="fignos">
<figure>
<img src="images/Fig2_Decalcification.png" alt="Figure 2: microangioCT-based visualization of the diaphysis of murine tibia before (A &amp; A’) and after decalcification with 10% EDTA (B &amp; B’). In A and A’ the tibia bone appears brighter and opaque due to higher X-ray absorption. In B and B’ the tibia bone appers transparent due to its lowered X-ray absorption after decalcification. Due to the decalcification, connecting vessels between the periostal vessels and the vessels of the bone marrow cavity (bmc) are more easily detectable (blue arrows in A vs. B). The visualization of the vessels within the medullar cavity (central sinus (CS)) is also improved. At the external surface of the tibia, supplying arteries are visible (arteria et vena nutricia (avn)). Scanning parameters (SKYSCAN 1172): acceleration voltage 59 kV, current 167 µA, 0.2 mm aluminum filter, 3.19 µm voxel size, 180 degrees scan, 0.1° rotation step, frame averaging 2." />
<figcaption aria-hidden="true"><span>Figure 2:</span> <strong>microangioCT-based visualization of the diaphysis of murine tibia before (A &amp; A’) and after decalcification with 10% EDTA (B &amp; B’).</strong>
In A and A’ the tibia bone appears brighter and opaque due to higher X-ray absorption.
In B and B’ the tibia bone appers transparent due to its lowered X-ray absorption after decalcification.
Due to the decalcification, connecting vessels between the periostal vessels and the vessels of the bone marrow cavity (bmc) are more easily detectable (blue arrows in A vs. B).
The visualization of the vessels within the medullar cavity (<em>central sinus</em> (CS)) is also improved.
At the external surface of the tibia, supplying arteries are visible (<em>arteria et vena nutricia</em> (avn)).
Scanning parameters (SKYSCAN 1172): acceleration voltage 59 kV, current 167 µA, 0.2 mm aluminum filter, 3.19 µm voxel size, 180 degrees scan, 0.1° rotation step, frame averaging 2.</figcaption>
</figure>
</div>
<h3 id="bone-metastatic-disease-model-microangioct">Bone metastatic disease model, microangioCT</h3>
<p>Assessment and 3D imaging of tumor vasculature is a challenging task, in almost every tumor model <span class="citation" data-cites="cNoYNu8B">[<a href="#ref-cNoYNu8B" role="doc-biblioref">32</a>]</span>.
Due to the intraosseous location of tumor lesions, 3D microvascular imaging is even more challenging in a bone metastatic disease model.
Since changes in bone and vasculature are believed to be of crucial importance for the progression of bone metastatic disease, simultaneous imaging of bone and vasculature is crucial.
The desire to simultaneously assess existing bone defects makes the aforementioned decalcification procedure highly unfavorable.</p>
<p>To verify the suitability of our microangioCT approach for this kind of experimental imaging we used the following murine bone metastatic disease model.
50000 PC3-M-Pro4Luc2 dTomato cells were injected into the tibia of 6 weeks old CB17SCID male mice as previously described <span class="citation" data-cites="C9ratYNO 16lhxZOpa">[<a href="#ref-C9ratYNO" role="doc-biblioref">33</a>,<a href="#ref-16lhxZOpa" role="doc-biblioref">34</a>]</span>.
The X-ray assessment (25 kV, 6 sec, Faxitron Bioptics, Tucson, Arizona, US) was conducted at days 7, 14, 21 and 28 after implantation to monitor the progression of the lesions.
Prior to terminal anesthesia and perfusion with µAngiofil as described above, the hind limb of interest was X-rayed using Faxitron Bioptics as a standard follow-up in this model (see insert in Fig. <a href="#fig:3">3</a>, Panel A).
The harvested and fixated murine hind limb was then imaged using a desktop microCT scanner SKYSCAN 1272 (Fig. <a href="#fig:3">3</a>).
The obtained dataset was reconstructed using NRecon Software v.1.7.4.2 (microCT Bruker, Kontich, Belgium) and visualized using CTvox Software v.3.3.1 (microCT Bruker, Kontich, Belgium).
The visualization unambiguously displays intratumoral vasculature and extensive defects in mineralized bone tissue.
Neighboring structures, such as the growth plate and the epiphysis or the menisci are also easily assessed (Fig. <a href="#fig:3">3</a>).</p>
<div id="fig:3" class="fignos">
<figure>
<img src="images/Fig3_tumor%20tibia.png" alt="Figure 3: microangioCT of a xenograft tumor implanted into murine tibia. A: Maximum intensity projection (MIP) of the three-dimensional dataset of the investigated hind limb segment indicated in the inserted X-ray image of the mouse prior harvesting. Inset: X-ray image prior to terminal anesthesia. B: Virtual section through the microangioCT dataset visualizing remarkable defects represented as holes (asterisks in B and C) in the tibial bone at the tumor site (encircled). C: A more deeply positioned virtual section displaying the inner surface of the diseased tibia. Besides irregularly patterned tumor vessels (in B &amp; C) further bony structures like growth plate or calcified parts of the medial meniscus (MM) are clearly distinguishable. Scanning parameters (SKYSCAN 1272): acceleration voltage 60 kV, current 166 µA, 0.2 mm aluminum filter, 1.65 µm voxel size, 360 degrees scan, 0.1° rotation step, frame averaging 3." />
<figcaption aria-hidden="true"><span>Figure 3:</span> <strong>microangioCT of a xenograft tumor implanted into murine tibia.</strong>
A: Maximum intensity projection (MIP) of the three-dimensional dataset of the investigated hind limb segment indicated in the inserted X-ray image of the mouse prior harvesting.
Inset: X-ray image prior to terminal anesthesia.
B: Virtual section through the microangioCT dataset visualizing remarkable defects represented as holes (asterisks in B and C) in the tibial bone at the tumor site (encircled).
C: A more deeply positioned virtual section displaying the inner surface of the diseased tibia.
Besides irregularly patterned tumor vessels (in B &amp; C) further bony structures like growth plate or calcified parts of the medial meniscus (MM) are clearly distinguishable.
Scanning parameters (SKYSCAN 1272): acceleration voltage 60 kV, current 166 µA, 0.2 mm aluminum filter, 1.65 µm voxel size, 360 degrees scan, 0.1° rotation step, frame averaging 3.</figcaption>
</figure>
</div>
<h3 id="microangioct-of-murine-mandibleteeth">microangioCT of murine mandible/teeth</h3>
<p>Murine mandibles and teeth are challenging samples due to the location of most of the vessels within the bone canals or in the proximity of the hard tissue and, correspondingly, the lack of larger bone-free volumes in which the vessels are easily distinguishable.
Nonetheless, with the improved perfusion protocol (see above) and a well visualized difference between the X-ray absorption levels of µAngiofil and mineralized bone tissue, we achieved appropriate imaging of the sample.
The microvasculature of murine mandible, periodontal ligament, and the teeth (even within their pulp chamber) can be clearly visualized without undergoing a decalcification procedure (Fig. <a href="#fig:4">4</a>).</p>
<div id="fig:4" class="fignos">
<figure>
<img src="images/Fig4_mouse%20teeth.png" alt="Figure 4: microangioCT of the vasculature of murine teeth. A: View onto the virtual horizontal section through the crowns of the murine teeth: The pulp chambers are visible, and the pulpal vessels are presented. Overview of the dataset in the inset at the upper left corner. B: Sagittal section through the mandible. The microvessels within the pulp cavities and root canals are distinct. C: A closer view onto the dentogingival plexus and periodontal ligament vessels (PLV). Scanning parameters (SKYSCAN 1172): accelerating voltage 80 kV, voxel size 1.0 µm, 0.2 mm aluminum filter, rotation step 0.1°, 360 degrees scan, frame averaging 4." />
<figcaption aria-hidden="true"><span>Figure 4:</span> <strong>microangioCT of the vasculature of murine teeth.</strong>
A: View onto the virtual horizontal section through the crowns of the murine teeth: The pulp chambers are visible, and the pulpal vessels are presented.
Overview of the dataset in the inset at the upper left corner.
B: Sagittal section through the mandible.
The microvessels within the pulp cavities and root canals are distinct.
C: A closer view onto the dentogingival plexus and periodontal ligament vessels (PLV).
Scanning parameters (SKYSCAN 1172): accelerating voltage 80 kV, voxel size 1.0 µm, 0.2 mm aluminum filter, rotation step 0.1°, 360 degrees scan, frame averaging 4.</figcaption>
</figure>
</div>
<h3 id="microangioct-of-mandible-in-a-large-animal-model-göttingen-minipig">microangioCT of mandible in a large animal model (Göttingen Minipig)</h3>
<p>Due to their similarity to humans in terms of anatomy, bone structure and bone remodeling, the Göttingen Minipig has become a well-accepted large animal model in preclinical dental and orofacial research <span class="citation" data-cites="Siy9vGW7 i45b7Cmp 1EXdAADA3">[<a href="#ref-Siy9vGW7" role="doc-biblioref">35</a>,<a href="#ref-i45b7Cmp" role="doc-biblioref">36</a>,<a href="#ref-1EXdAADA3" role="doc-biblioref">37</a>]</span>.
The stricter legal requirements regarding primates and the societal resistance against the use of companion animals, such as dogs and cats, have further enhanced the usage of Göttingen minipigs as animal models <span class="citation" data-cites="i45b7Cmp">[<a href="#ref-i45b7Cmp" role="doc-biblioref">36</a>]</span>.</p>
<p>Göttingen minipigs (Ellegaard Göttingen Minipig, Dalmose, Denmark) were intramuscularly anesthetized (25-35 mg/kg, Dexdomitor; Orion Pharma Animal Health and 50-70 mg/kg, Zoletil 100 Vet, Virbac) and heparinized with 300 IE/kg intravenously (Heparin LEO, LEO Pharma).
After heparin infusion, the pigs were euthanized with an intravenous dose (100 mg/kg) of pentobarbital (Euthanimal vet, VM Pharma).
<em>A.
carotis externa</em> was accessed by blunt dissection through tissue of the ventral neck and cannulated (BD Venflon, 17G).
After washing out the blood with PBS, the corresponding head side was selectively perfused with µAngiofil through the arterial tree.
The success of the perfusion has been visually controlled equivalent to the previously described perfusion in mice.
After polymerization of µAngiofil (~30 min), the mandible was excised and fixated in 4% PFA solution.
They were later scanned using SKYSCAN 2214 or SKYSCAN 1273, and the datasets reconstructed and visualized using NRecon v.1.7.4.2 and CTvox Software v.3.3.1, correspondingly (both microCT Bruker, Kontich, Belgium).
Our approach provides excellent imaging of the whole mandible vasculature down to the vessels within the root canals and pulp chamber (Fig. <a href="#fig:5">5</a>).</p>
<div id="fig:5" class="fignos">
<figure>
<img src="images/Fig5_Vreni%20jaw%20minipig.png" alt="Figure 5: microangioCT of the minipig mandibula. Panel A displays the visualization of a right minipig hemimandible: the vasculature at the bone surface is clearly visible. The framed area in A marks the subvolume represented in panel B at higher magnification. Panel C displays the transverse section marked in panel B: the pulp chamber and root canal with the corresponding vessels are unambiguously visualized. Scanning parameters for images in panel A (SKYSCAN 1273) accelerating voltage 100 kV, current 80 µA, 9 µm voxel size, 1 mm aluminum and 0.2 mm copper filter, rotation step 0.15°, 360 degrees scan, frame averaging of 5. Scanning parameters for images in panels B and C (SKYSCAN 2214): accelerating voltage 100 kV, current 100 µA, 8 µm voxel size, 1 mm copper filter, 0.18° rotation step, 360 degrees scan, frame averaging 4. Due to the voxel size of 8-9 µm, microvessels with diameter of 40 µm or less cannot be visualized in such large samples." />
<figcaption aria-hidden="true"><span>Figure 5:</span> <strong>microangioCT of the minipig mandibula.</strong>
Panel A displays the visualization of a right minipig hemimandible: the vasculature at the bone surface is clearly visible.
The framed area in A marks the subvolume represented in panel B at higher magnification.
Panel C displays the transverse section marked in panel B: the pulp chamber and root canal with the corresponding vessels are unambiguously visualized.
Scanning parameters for images in panel A (SKYSCAN 1273) accelerating voltage 100 kV, current 80 µA, 9 µm voxel size, 1 mm aluminum and 0.2 mm copper filter, rotation step 0.15°, 360 degrees scan, frame averaging of 5.
Scanning parameters for images in panels B and C (SKYSCAN 2214): accelerating voltage 100 kV, current 100 µA, 8 µm voxel size, 1 mm copper filter, 0.18° rotation step, 360 degrees scan, frame averaging 4.
Due to the voxel size of 8-9 µm, microvessels with diameter of 40 µm or less cannot be visualized in such large samples.</figcaption>
</figure>
</div>
<p>As already mentioned, the Göttingen Minipig has become a very popular large animal model in dental research and is often used also for implantology studies.
After researchers in the field realized that osseointegration is influenced by angiogenesis, there is understanding that angiogenesis and, correspondingly, vascular supply of the peri-implant tissue should be assessed as thoroughly as possible and preferably in 3D.
So far, the only reliable approach to assess the vascular supply remains histology, which is limited to single two-dimensional sections.
Based on the findings in those single histological sections, researchers try to assess the vascularization of the peri-implant bone.
Even for the microCT approach, it is a challenging task due to the presence of metal parts and, therefore, multiple objects with high density within the samples.
The decalcification step could reduce the density of the sample but would inevitably lead to the loss of information on the bone microstructure making the simultaneous assessment of the bone and vessels impossible <span class="citation" data-cites="117R9ULnv">[<a href="#ref-117R9ULnv" role="doc-biblioref">9</a>]</span>.
Another obstacle is to have a contrast agent that can be distinctly separated from both the metal implants and the mineralized tissue.
With µAngiofil it is easily possible to visualize and distinguish between soft tissue, bone tissue, contrast agent-filled vessels as well as metal implants according to their gray values in the histogram (Fig. <a href="#fig:6">6</a>, Panel C).
Such imaging does not require bone decalcification and enables a straightforward segmentation of features of interest without cumbersome post-processing (Fig. <a href="#fig:6">6</a>, Panel D).</p>
<div id="fig:6" class="fignos">
<figure>
<img src="images/Fig6_Segmentation.png" alt="Figure 6: microangioCT of the peri-implant vasculature of a minipig mandibula. Panel A displays a MIP image of the minipig mandible with 4 metal implants and µAngiofil–perfused vessels. Panel B: virtual transversal section through an implant within the mandible. Colored circles mark structures with distinguishably different grey values. See the histogram distribution and corresponding peaks with colors legend in panel C. Such differences in grey levels allow a straightforward segmentation of the structures of interest as displayed in panel D. Scanning parameters (SKYSCAN 2214): accelerating voltage 100 kV, voxel size 8.0 µm, 1 mm copper filter, rotation step 0.18°, 360 degrees scan, frame averaging 4." />
<figcaption aria-hidden="true"><span>Figure 6:</span> <strong>microangioCT of the peri-implant vasculature of a minipig mandibula.</strong>
Panel A displays a MIP image of the minipig mandible with 4 metal implants and µAngiofil–perfused vessels.
Panel B: virtual transversal section through an implant within the mandible.
Colored circles mark structures with distinguishably different grey values.
See the histogram distribution and corresponding peaks with colors legend in panel C.
Such differences in grey levels allow a straightforward segmentation of the structures of interest as displayed in panel D.
Scanning parameters (SKYSCAN 2214): accelerating voltage 100 kV, voxel size 8.0 µm, 1 mm copper filter, rotation step 0.18°, 360 degrees scan, frame averaging 4.</figcaption>
</figure>
</div>
<h2 class="page_break_before" id="discussion">Discussion</h2>
<p>There is an understanding in the bone biology, tissue engineering and related research fields, that adequate vascularization is a prerequisite for successful bone formation and regeneration as well as osseointegration of biomaterials <span class="citation" data-cites="lVl4gVyN Ha1wHU4s zB5E7Obz zXoIXG3G 16mB4VUpP">[<a href="#ref-zXoIXG3G" role="doc-biblioref">1</a>,<a href="#ref-lVl4gVyN" role="doc-biblioref">38</a>,<a href="#ref-Ha1wHU4s" role="doc-biblioref">39</a>,<a href="#ref-zB5E7Obz" role="doc-biblioref">40</a>,<a href="#ref-16mB4VUpP" role="doc-biblioref">41</a>]</span>.
Without high-resolution imaging, the interplay between vasculature and bone tissue cannot be adequately studied, making it impossible to influence or even control these processes.
During the last decades, microCT gained recognition as a non-destructive 3D imaging technique for bone morphology <span class="citation" data-cites="KjvwGWK2">[<a href="#ref-KjvwGWK2" role="doc-biblioref">14</a>]</span>.
It is considered the only available approach for 3D imaging of an intact bone-implant interface <span class="citation" data-cites="11349lK8v">[<a href="#ref-11349lK8v" role="doc-biblioref">21</a>]</span>.
Unfortunately, the microCT-based distinction of the vessels from other soft tissue structures is not possible without intravascular casting or instillation of contrast agents.
Moreover, although enabling the visualization of the vasculature, most such casting or contrast agents are difficult to distinguish from the non-demineralized bone tissue due to the lack of remarkable differences in their X-ray absorption levels <span class="citation" data-cites="13YRlKdu2">[<a href="#ref-13YRlKdu2" role="doc-biblioref">16</a>]</span>.
For example, in case with Microfil, the mentioned lack of contrast requires decalcification of the samples before the bone vasculature can be properly visualized with a microCT scanner <span class="citation" data-cites="117R9ULnv 13YRlKdu2">[<a href="#ref-117R9ULnv" role="doc-biblioref">9</a>,<a href="#ref-13YRlKdu2" role="doc-biblioref">16</a>]</span>.
Some studies have described that barium sulfate suspension could be used for direct simultaneous imaging of vasculature and bone <span class="citation" data-cites="168Ee16HU 1ByuShsGE">[<a href="#ref-168Ee16HU" role="doc-biblioref">15</a>,<a href="#ref-1ByuShsGE" role="doc-biblioref">42</a>]</span>.
Many studies report higher viscosity of this agent in comparison to Microfil or µAngiofil, though which leads to inconsistent vascular filling or even the inability to perfuse fine capillaries <span class="citation" data-cites="Ror9Uhfa 1ByuShsGE">[<a href="#ref-Ror9Uhfa" role="doc-biblioref">19</a>,<a href="#ref-1ByuShsGE" role="doc-biblioref">42</a>]</span>.
The hereby described approach with µAngiofil provides more complete vascular filling <span class="citation" data-cites="s9aC9H1e Ror9Uhfa">[<a href="#ref-Ror9Uhfa" role="doc-biblioref">19</a>,<a href="#ref-s9aC9H1e" role="doc-biblioref">28</a>]</span> and is suitable for the visualization of the microvasculature within bone tissue with or without decalcification (Fig. <a href="#fig:2">2</a>).
If a subsequent histological evaluation is desired, the developed decalcifying protocols enable correlative imaging of the same sample with preserved intravascular contrast agent <span class="citation" data-cites="iM9i6gkJ">[<a href="#ref-iM9i6gkJ" role="doc-biblioref">31</a>]</span>.</p>
<p>Distinct differences in the attenuation between the µAngiofil and bone tissue allow for assessing vascularization and bone growth without the need to make two scans (before and after decalcification) and register them, saving labor, scanning time and without damaging the sample composition.</p>
<p>The microangioCT approach we present here can be used for simultaneous visualization of the hard (and eventually soft) tissues and their vascularization in small and large animal models.
One of the noteworthy possible applications of this approach is simultaneous visualization of the bone microarchitecture and microvasculature of the bone metastases in the murine xenograft tumor model (Fig. <a href="#fig:3">3</a>).
Using such an approach, the response to specific treatments, including bone-targeted and antiangiogenic therapies, can be assessed within the same sample and in 3D, followed by histological examination if needed <span class="citation" data-cites="15Cb0RCtH">[<a href="#ref-15Cb0RCtH" role="doc-biblioref">43</a>]</span>.
The presented method could offer additional insights into the interplay between angiogenesis and bone growth, bone lysis, and bone turnover, which have not been fully elucidated yet.
The results will be crucial for selecting potential drug candidates and proper treatment decision and efficacy prediction.
The decalcification step would be very disadvantageous for studying bone metastatic models.
Such a step would hinder the thorough assessment of pathological processes in bone tissue.
These processes may lead to so-called skeletal-related-events associated with shortened survival and deterioration of quality of life <span class="citation" data-cites="D7D7Wyla">[<a href="#ref-D7D7Wyla" role="doc-biblioref">44</a>]</span>, and should therefore be avoided.</p>
<p>In dental research, preclinical models can be divided into small and large animals.
Small animal models, particularly mice and rats, are highly popular, due to their practical size and cost-effectiveness.
Small animal models, such as mice and rats, are highly popular due to their practical size and reduced cost.
Moreover, there is a variety of genetically modified lines available which could facilitate imaging if using corresponding imaging modalities <span class="citation" data-cites="11349lK8v">[<a href="#ref-11349lK8v" role="doc-biblioref">21</a>]</span>.
The microCT scan of correspondingly small samples perfused with µAngiofil can be performed with a voxel size around 1 µm, providing excellent detail resolution (Fig. <a href="#fig:4">4</a>).</p>
<p>Nonetheless, the small size of hemimandible samples and availability of the transgenic lines would allow for a recently published imaging approach <span class="citation" data-cites="11349lK8v">[<a href="#ref-11349lK8v" role="doc-biblioref">21</a>]</span> in the murine models.
This tissue-clearing-based approach could be a viable option for the visualization of the peri-implant tissues and vasculature in small animal models.
There are limitations of such a tissue-clearing-based imaging approach, namely
i) differential shrinkage among soft and hard tissues, leading to anisotropic distortion in samples where both tissue types (plus eventual metal implant) are present;
ii) limited depth (around 800 µm in a mouse model <span class="citation" data-cites="11349lK8v">[<a href="#ref-11349lK8v" role="doc-biblioref">21</a>]</span>) due to the challenges of achieving complete transparency of bone tissue.
These limitations become even more critical in a big animal model with metal implants.
Large anisotropic distortion of the sample may lead to the alteration of the common site of interest, namely the implant-tissue interface.
Furthermore, the limitation of the visible depth makes such an approach impracticable for usage in large animal models.</p>
<p>The microangioCT approach does not have such limitations and can be easily applied for visualization of hemimandible and its vascularization in a large animal model like Göttingen minipig (Fig. <a href="#fig:5">5</a>).
µAngiofil has noticeably different attenuation than mineralized bone and it is useful when scanning the samples, which contain high-density materials, like metals.
It is thus possible to visualize and distinguish soft tissue, bone tissue, vessels filled with the contrast agent as well as metal implants according to their gray values in the histogram (Fig. <a href="#fig:6">6</a>).</p>
<p>To the best of our knowledge, this is the first study demonstrating non-destructive 3D imaging of the microvasculature of bone in the proximity of metal objects/implants in a large animal model.</p>
<h3 id="limitations-of-the-approach">Limitations of the approach</h3>
<p>The most significant limitation of the suggested approach is the general limitation of microCT imaging, namely, the correlation between the resolution and the sample size.
If the sample is only a few millimeters in diameter, the voxel size may be as small as 1 µm or less enabling visualization of the microvasculature down to the capillary bed <span class="citation" data-cites="TYAmre95 s9aC9H1e ijqzbOI6 iM9i6gkJ">[<a href="#ref-s9aC9H1e" role="doc-biblioref">28</a>,<a href="#ref-ijqzbOI6" role="doc-biblioref">29</a>,<a href="#ref-TYAmre95" role="doc-biblioref">30</a>,<a href="#ref-iM9i6gkJ" role="doc-biblioref">31</a>]</span>.
In case of large samples, like minipig hemimandible, the achievable voxel size is around 10 times larger (≥8 µm), and it is therefore, not possible to reach that level of resolution.
In such samples the finest microvessels with diameter under 40 µm cannot be accurately visualized (see Fig. <a href="#fig:5">5</a> &amp; <a href="#fig:6">6</a>).</p>
<p>Nonetheless, the presented approach is, according to our knowledge, the best what can be achieved with the modern microCT scanners and is a big step forward for qualitative and quantitative imaging in tissue engineering and implantology research as well as many other related research fields.</p>
<h2 class="page_break_before" id="author-contributions">Author Contributions</h2>
<p><a href="https://credit.niso.org/">Contributor Roles Taxonomy</a>, as defined by the <a href="https://credit.niso.org/contributor-roles-defined/">National Information Standards Organization</a>.</p>
<table>
<colgroup>
<col style="width: 50%" />
<col style="width: 50%" />
</colgroup>
<thead>
<tr class="header">
<th>Author</th>
<th>Contributions</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>David Haberthür</td>
<td>Data curation; Formal analysis; Investigation; Project administration; Software; Validation; Visualization; Writing – original draft; Writing – review &amp; editing</td>
</tr>
<tr class="even">
<td>Oleksiy-Zakhar Khoma</td>
<td>Data curation; Formal analysis; Investigation; Project administration; Software; Validation; Visualization; Writing – original draft; Writing – review &amp; editing</td>
</tr>
<tr class="odd">
<td>Tim Hoessly</td>
<td>Data curation; Investigation; Writing – review &amp; editing</td>
</tr>
<tr class="even">
<td>Eugenio Zoni</td>
<td>Resources; Writing – review &amp; editing</td>
</tr>
<tr class="odd">
<td>Marianna Kruithof-de Julio</td>
<td>Resources; Writing – review &amp; editing</td>
</tr>
<tr class="even">
<td>Stewart D. Ryan</td>
<td>Resources; Writing – review &amp; editing</td>
</tr>
<tr class="odd">
<td>Myriam Grunewald</td>
<td>Resources; Writing – review &amp; editing</td>
</tr>
<tr class="even">
<td>Benjamin Bellón</td>
<td>Methodology; Writing – review &amp; editing</td>
</tr>
<tr class="odd">
<td>Rebecca Sandgren</td>
<td>Resources; Writing – review &amp; editing</td>
</tr>
<tr class="even">
<td>Benjamin E. Pippenger</td>
<td>Methodology; Writing – review &amp; editing</td>
</tr>
<tr class="odd">
<td>Dieter Bosshardt</td>
<td>Methodology; Writing – review &amp; editing</td>
</tr>
<tr class="even">
<td>Valentin Djonov</td>
<td>Conceptualization; Funding acquisition; Supervision; Writing – review &amp; editing</td>
</tr>
<tr class="odd">
<td>Ruslan Hlushchuk</td>
<td>Conceptualization; Funding acquisition; Investigation; Methodology; Project administration; Supervision; Validation; Writing – original draft; Writing – review &amp; editing</td>
</tr>
</tbody>
</table>
<h2 class="page_break_before" id="conflicts-of-interest">Conflicts of interest</h2>
<table>
<thead>
<tr class="header">
<th>Author</th>
<th>Conflict</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Benjamin Bellón</td>
<td>Employed by a company producing dental implants</td>
</tr>
<tr class="even">
<td>Benjamin E. Pippenger</td>
<td>Employed by a company producing dental implants</td>
</tr>
</tbody>
</table>
<h2 id="acknowledgments">Acknowledgments</h2>
<h2 id="supplementary-materials">Supplementary Materials</h2>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references csl-bib-body" role="doc-bibliography">
<div id="ref-zXoIXG3G" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">1. </div><div class="csl-right-inline"><strong>Long-term cell-mediated protein release from calcium phosphate ceramics</strong> <div class="csl-block">Ellen Wernike, Willy Hofstetter, Yuelian Liu, Gang Wu, Hans-Jörg Sebald, Daniel Wismeijer, Ernst B Hunziker, Klaus-Arno Siebenrock, Frank M Klenke</div> <em>Journal of Biomedical Materials Research Part A</em> (2009) <a href="https://doi.org/bmqbc8">https://doi.org/bmqbc8</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/jbm.a.32411">10.1002/jbm.a.32411</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19195029">19195029</a></div></div>
</div>
<div id="ref-AHsHFIUt" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">2. </div><div class="csl-right-inline"><strong>Structure and quantification of microvascularisation within mouse long bones: What and how should we measure?</strong> <div class="csl-block">Bernard Roche, Valentin David, Arnaud Vanden-Bossche, Françoise Peyrin, Luc Malaval, Laurence Vico, Marie-Hélène Lafage-Proust</div> <em>Bone</em> (2012-01) <a href="https://doi.org/bpvvpm">https://doi.org/bpvvpm</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.bone.2011.09.051">10.1016/j.bone.2011.09.051</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22019874">22019874</a></div></div>
</div>
<div id="ref-1FaKEBtU8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">3. </div><div class="csl-right-inline"><strong>Bone Vascularization in Normal and Disease Conditions</strong> <div class="csl-block">Christian Carulli, Massimo Innocenti, Maria Luisa Brandi</div> <em>Frontiers in Endocrinology</em> (2013) <a href="https://doi.org/gr738c">https://doi.org/gr738c</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fendo.2013.00106">10.3389/fendo.2013.00106</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23986744">23986744</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3752619">PMC3752619</a></div></div>
</div>
<div id="ref-Vu36g0cK" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">4. </div><div class="csl-right-inline"><strong>State-of-the-art techniques for imaging the vascular microenvironment in craniofacial bone tissue engineering applications</strong> <div class="csl-block">Yunke Ren, Janaka Senarathna, Warren L Grayson, Arvind P Pathak</div> <em>American Journal of Physiology-Cell Physiology</em> (2022-11-01) <a href="https://doi.org/gr7378">https://doi.org/gr7378</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1152/ajpcell.00195.2022">10.1152/ajpcell.00195.2022</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/36189973">36189973</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9829486">PMC9829486</a></div></div>
</div>
<div id="ref-yZTJBX1B" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">5. </div><div class="csl-right-inline"><strong>The role of vasculature in bone development, regeneration and proper systemic functioning</strong> <div class="csl-block">Joanna Filipowska, Krzysztof A Tomaszewski, Łukasz Niedźwiedzki, Jerzy A Walocha, Tadeusz Niedźwiedzki</div> <em>Angiogenesis</em> (2017-02-13) <a href="https://doi.org/gbn8bj">https://doi.org/gbn8bj</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s10456-017-9541-1">10.1007/s10456-017-9541-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28194536">28194536</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5511612">PMC5511612</a></div></div>
</div>
<div id="ref-dBkW39I8" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">6. </div><div class="csl-right-inline"><strong>Dual release of growth factor from nanocomposite fibrous scaffold promotes vascularisation and bone regeneration in rat critical sized calvarial defect</strong> <div class="csl-block">Shruthy Kuttappan, Dennis Mathew, Jun-ichiro Jo, Ryusuke Tanaka, Deepthy Menon, Takuya Ishimoto, Takayoshi Nakano, Shantikumar V Nair, Manitha B Nair, Yasuhiko Tabata</div> <em>Acta Biomaterialia</em> (2018-09) <a href="https://doi.org/gr737v">https://doi.org/gr737v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.actbio.2018.07.050">10.1016/j.actbio.2018.07.050</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30067947">30067947</a></div></div>
</div>
<div id="ref-Es64ZQO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">7. </div><div class="csl-right-inline"><strong>Angiogenesis involvement by octacalcium phosphate-gelatin composite-driven bone regeneration in rat calvaria critical-sized defect</strong> <div class="csl-block">Tsuyoshi Kurobane, Yukari Shiwaku, Takahisa Anada, Ryo Hamai, Kaori Tsuchiya, Kazuyoshi Baba, Masahiro Iikubo, Tetsu Takahashi, Osamu Suzuki</div> <em>Acta Biomaterialia</em> (2019-04) <a href="https://doi.org/gr737w">https://doi.org/gr737w</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.actbio.2019.02.021">10.1016/j.actbio.2019.02.021</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30776505">30776505</a></div></div>
</div>
<div id="ref-5mgeUNd6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">8. </div><div class="csl-right-inline"><strong>It Takes Two to Tango: Coupling of Angiogenesis and Osteogenesis for Bone Regeneration</strong> <div class="csl-block">Andrea Grosso, Maximilian G Burger, Alexander Lunger, Dirk J Schaefer, Andrea Banfi, Nunzia Di Maggio</div> <em>Frontiers in Bioengineering and Biotechnology</em> (2017-11-03) <a href="https://doi.org/gckgs4">https://doi.org/gckgs4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fbioe.2017.00068">10.3389/fbioe.2017.00068</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29164110">29164110</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5675838">PMC5675838</a></div></div>
</div>
<div id="ref-117R9ULnv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">9. </div><div class="csl-right-inline"><strong>Simultaneous visualisation of calcified bone microstructure and intracortical vasculature using synchrotron X-ray phase contrast-enhanced tomography</strong> <div class="csl-block">Juan A Núñez, Alice Goring, Eric Hesse, Philipp J Thurner, Philipp Schneider, Claire E Clarkin</div> <em>Scientific Reports</em> (2017-10-16) <a href="https://doi.org/gcfxxb">https://doi.org/gcfxxb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41598-017-13632-5">10.1038/s41598-017-13632-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29038597">29038597</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5643345">PMC5643345</a></div></div>
</div>
<div id="ref-DiX3YTwQ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">10. </div><div class="csl-right-inline"><strong>Quantitative 3D imaging of the cranial microvascular environment at single-cell resolution</strong> <div class="csl-block">Alexandra N Rindone, Xiaonan Liu, Stephanie Farhat, Alexander Perdomo-Pantoja, Timothy F Witham, Daniel L Coutu, Mei Wan, Warren L Grayson</div> <em>Nature Communications</em> (2021-10-28) <a href="https://doi.org/gr7373">https://doi.org/gr7373</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41467-021-26455-w">10.1038/s41467-021-26455-w</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34711819">34711819</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8553857">PMC8553857</a></div></div>
</div>
<div id="ref-X0aG1K2m" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">11. </div><div class="csl-right-inline"><strong>Investigation of Postnatal Craniofacial Bone Development with Tissue Clearing-Based Three-Dimensional Imaging</strong> <div class="csl-block">Wenjing Luo, Yating Yi, Dian Jing, Shiwen Zhang, Yi Men, Woo-Ping Ge, Hu Zhao</div> <em>Stem Cells and Development</em> (2019-10-01) <a href="https://doi.org/gr7375">https://doi.org/gr7375</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1089/scd.2019.0104">10.1089/scd.2019.0104</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31392933">31392933</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6767869">PMC6767869</a></div></div>
</div>
<div id="ref-33Vq8bQx" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">12. </div><div class="csl-right-inline"><strong>Vascularization Strategies for Tissue Engineering</strong> <div class="csl-block">Michael Lovett, Kyongbum Lee, Aurelie Edwards, David L Kaplan</div> <em>Tissue Engineering Part B: Reviews</em> (2009-09) <a href="https://doi.org/fsjhcx">https://doi.org/fsjhcx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1089/ten.teb.2009.0085">10.1089/ten.teb.2009.0085</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19496677">19496677</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2817665">PMC2817665</a></div></div>
</div>
<div id="ref-1FUMXJVtA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">13. </div><div class="csl-right-inline"><strong>Vascularization in Bone Tissue Engineering Constructs</strong> <div class="csl-block">Ángel E Mercado-Pagán, Alexander M Stahl, Yaser Shanjani, Yunzhi Yang</div> <em>Annals of Biomedical Engineering</em> (2015-01-24) <a href="https://doi.org/f66z3g">https://doi.org/f66z3g</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s10439-015-1253-3">10.1007/s10439-015-1253-3</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25616591">25616591</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4979539">PMC4979539</a></div></div>
</div>
<div id="ref-KjvwGWK2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">14. </div><div class="csl-right-inline"><strong>Modalities for Visualization of Cortical Bone Remodeling: The Past, Present, and Future</strong> <div class="csl-block">Kimberly D Harrison, David ML Cooper</div> <em>Frontiers in Endocrinology</em> (2015-08-11) <a href="https://doi.org/gr738d">https://doi.org/gr738d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3389/fendo.2015.00122">10.3389/fendo.2015.00122</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26322017">26322017</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4531299">PMC4531299</a></div></div>
</div>
<div id="ref-168Ee16HU" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">15. </div><div class="csl-right-inline"><strong>Simultaneous 3D visualization and quantification of murine bone and bone vasculature using micro-computed tomography and vascular replica</strong> <div class="csl-block">Philipp Schneider, Thomas Krucker, Eric Meyer, Alexandra Ulmann-Schuler, Bruno Weber, Marco Stampanoni, Ralph Müller</div> <em>Microscopy Research and Technique</em> (2009-09) <a href="https://doi.org/ftspdd">https://doi.org/ftspdd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/jemt.20720">10.1002/jemt.20720</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19360841">19360841</a></div></div>
</div>
<div id="ref-13YRlKdu2" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">16. </div><div class="csl-right-inline"><strong>Microcomputed Tomography Characterization of Neovascularization in Bone Tissue Engineering Applications</strong> <div class="csl-block">Simon Young, James D Kretlow, Charles Nguyen, Alex G Bashoura, LScott Baggett, John A Jansen, Mark Wong, Antonios G Mikos</div> <em>Tissue Engineering Part B: Reviews</em> (2008-09) <a href="https://doi.org/b5snxg">https://doi.org/b5snxg</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1089/ten.teb.2008.0153">10.1089/ten.teb.2008.0153</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18657028">18657028</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2761680">PMC2761680</a></div></div>
</div>
<div id="ref-DWiXWNoA" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">17. </div><div class="csl-right-inline"><strong>New polyurethane-based material for vascular corrosion casting with improved physical and imaging characteristics</strong> <div class="csl-block">Thomas Krucker, Axel Lang, Eric P Meyer</div> <em>Microscopy Research and Technique</em> (2006) <a href="https://doi.org/d6dwtd">https://doi.org/d6dwtd</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/jemt.20263">10.1002/jemt.20263</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16456839">16456839</a></div></div>
</div>
<div id="ref-KJUjdwi7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">18. </div><div class="csl-right-inline"><strong>Novel methods for microCT-based analyses of vasculature in the renal cortex reveal a loss of perfusable arterioles and glomeruli in eNOS-/- mice</strong> <div class="csl-block">Daniel S Perrien, Mohamed A Saleh, Keiko Takahashi, Meena S Madhur, David G Harrison, Raymond C Harris, Takamune Takahashi</div> <em>BMC Nephrology</em> (2016-03-02) <a href="https://doi.org/gr738b">https://doi.org/gr738b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1186/s12882-016-0235-5">10.1186/s12882-016-0235-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26936597">26936597</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4776352">PMC4776352</a></div></div>
</div>
<div id="ref-Ror9Uhfa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">19. </div><div class="csl-right-inline"><strong>A Review of Ex Vivo X-ray Microfocus Computed Tomography-Based Characterization of the Cardiovascular System</strong> <div class="csl-block">Lisa Leyssens, Camille Pestiaux, Greet Kerckhofs</div> <em>International Journal of Molecular Sciences</em> (2021-03-23) <a href="https://doi.org/gr738g">https://doi.org/gr738g</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3390/ijms22063263">10.3390/ijms22063263</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33806852">33806852</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8004599">PMC8004599</a></div></div>
</div>
<div id="ref-1A0HsmRho" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">20. </div><div class="csl-right-inline"><strong>Development of barium‐based low viscosity contrast agents for micro CT vascular casting: Application to 3D visualization of the adult mouse cerebrovasculature</strong> <div class="csl-block">Sung‐Ha Hong, Alexander M Herman, Jessica M Stephenson, Ting Wu, Ali N Bahadur, Alan R Burns, Sean P Marrelli, Joshua D Wythe</div> <em>Journal of Neuroscience Research</em> (2019-10-19) <a href="https://doi.org/gr737r">https://doi.org/gr737r</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1002/jnr.24539">10.1002/jnr.24539</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31630455">31630455</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8063604">PMC8063604</a></div></div>
</div>
<div id="ref-11349lK8v" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">21. </div><div class="csl-right-inline"><strong>3‐dimensional visualization of implant‐tissue interface with the polyethylene glycol associated solvent system tissue clearing method</strong> <div class="csl-block">Yating Yi, Yi Men, Dian Jing, Wenjing Luo, Shiwen Zhang, Jian Q Feng, Jin Liu, Woo‐Ping Ge, Jun Wang, Hu Zhao</div> <em>Cell Proliferation</em> (2019-02-03) <a href="https://doi.org/gr7376">https://doi.org/gr7376</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/cpr.12578">10.1111/cpr.12578</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30714253">30714253</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6536405">PMC6536405</a></div></div>
</div>
<div id="ref-rCXuqa0O" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">22. </div><div class="csl-right-inline"><strong>Multifunctional coatings to simultaneously promote osseointegration and prevent infection of orthopaedic implants</strong> <div class="csl-block">Jordan Raphel, Mark Holodniy, Stuart B Goodman, Sarah C Heilshorn</div> <em>Biomaterials</em> (2016-04) <a href="https://doi.org/f8cm8j">https://doi.org/f8cm8j</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.biomaterials.2016.01.016">10.1016/j.biomaterials.2016.01.016</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26851394">26851394</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4883578">PMC4883578</a></div></div>
</div>
<div id="ref-137VU0bph" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">23. </div><div class="csl-right-inline"><strong>Differential effect of hydroxyapatite nano-particle versus nano-rod decorated titanium micro-surface on osseointegration</strong> <div class="csl-block">Long Bai, Yanlian Liu, Zhibin Du, Zeming Weng, Wei Yao, Xiangyu Zhang, Xiaobo Huang, Xiaohong Yao, Ross Crawford, Ruiqiang Hang, … Yin Xiao</div> <em>Acta Biomaterialia</em> (2018-08) <a href="https://doi.org/gm7sqx">https://doi.org/gm7sqx</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.actbio.2018.06.023">10.1016/j.actbio.2018.06.023</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29908975">29908975</a></div></div>
</div>
<div id="ref-5CxAuEpv" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">24. </div><div class="csl-right-inline"><strong>Anti-VEGFs hinder bone healing and implant osseointegration in rat tibiae</strong> <div class="csl-block">Ahmed Ebraheem Al Subaie, Hazem Eimar, Mohamed-Nur Abdallah, Robert Durand, Jocelyne Feine, Faleh Tamimi, Elham Emami</div> <em>Journal of Clinical Periodontology</em> (2015-07) <a href="https://doi.org/f7kf9t">https://doi.org/f7kf9t</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/jcpe.12424">10.1111/jcpe.12424</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26073407">26073407</a></div></div>
</div>
<div id="ref-uAjlWLJe" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">25. </div><div class="csl-right-inline"><strong>Neural peptide promotes the angiogenesis and osteogenesis around oral implants</strong> <div class="csl-block">B Wang, B Wu, Y Jia, Y Jiang, Y Yuan, Y Man, L Xiang</div> <em>Cellular Signalling</em> (2021-03) <a href="https://doi.org/gr897c">https://doi.org/gr897c</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.cellsig.2020.109873">10.1016/j.cellsig.2020.109873</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33285241">33285241</a></div></div>
</div>
<div id="ref-jhdsbJV3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">26. </div><div class="csl-right-inline"><strong>VascuViz: a multimodality and multiscale imaging and visualization pipeline for vascular systems biology</strong> <div class="csl-block">Akanksha Bhargava, Benjamin Monteagudo, Priyanka Kushwaha, Janaka Senarathna, Yunke Ren, Ryan C Riddle, Manisha Aggarwal, Arvind P Pathak</div> <em>Nature Methods</em> (2022-02) <a href="https://doi.org/gr897d">https://doi.org/gr897d</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/s41592-021-01363-5">10.1038/s41592-021-01363-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35145319">35145319</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8842955">PMC8842955</a></div></div>
</div>
<div id="ref-nC0ZZVgL" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">27. </div><div class="csl-right-inline"><strong>Counteracting age-related VEGF signaling insufficiency promotes healthy aging and extends life span</strong> <div class="csl-block">M Grunewald, S Kumar, H Sharife, E Volinsky, A Gileles-Hillel, T Licht, A Permyakova, L Hinden, S Azar, Y Friedmann, … E Keshet</div> <em>Science</em> (2021-07-30) <a href="https://doi.org/gmc94x">https://doi.org/gmc94x</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1126/science.abc8479">10.1126/science.abc8479</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/34326210">34326210</a></div></div>
</div>
<div id="ref-s9aC9H1e" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">28. </div><div class="csl-right-inline"><strong>Innovative high-resolution microCT imaging of animal brain vasculature</strong> <div class="csl-block">Ruslan Hlushchuk, David Haberthür, Petr Soukup, Sebastien F Barré, Oleksiy-Zakhar Khoma, Johannes Schittny, Neda Haghayegh Jahromi, Audrey Bouchet, Britta Engelhardt, Valentin Djonov</div> <em>Brain Structure and Function</em> (2020-10-31) <a href="https://doi.org/gr737t">https://doi.org/gr737t</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/s00429-020-02158-8">10.1007/s00429-020-02158-8</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33128675">33128675</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7674347">PMC7674347</a></div></div>
</div>
<div id="ref-ijqzbOI6" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">29. </div><div class="csl-right-inline"><strong>Cutting-edge microangio-CT: new dimensions in vascular imaging and kidney morphometry</strong> <div class="csl-block">Ruslan Hlushchuk, Cédric Zubler, Sébastien Barré, Carlos Correa Shokiche, Laura Schaad, Raphael Röthlisberger, Monika Wnuk, Christoph Daniel, Oleksiy Khoma, Stefan A Tschanz, … Valentin Djonov</div> <em>American Journal of Physiology-Renal Physiology</em> (2018-03-01) <a href="https://doi.org/gdbw3b">https://doi.org/gdbw3b</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1152/ajprenal.00099.2017">10.1152/ajprenal.00099.2017</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29167169">29167169</a></div></div>
</div>
<div id="ref-TYAmre95" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">30. </div><div class="csl-right-inline"><strong>Ex vivo microangioCT: Advances in microvascular imaging</strong> <div class="csl-block">Ruslan Hlushchuk, David Haberthür, Valentin Djonov</div> <em>Vascular Pharmacology</em> (2019-01) <a href="https://doi.org/ghgd4s">https://doi.org/ghgd4s</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.vph.2018.09.003">10.1016/j.vph.2018.09.003</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30248380">30248380</a></div></div>
</div>
<div id="ref-iM9i6gkJ" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">31. </div><div class="csl-right-inline"><strong>Correlative Imaging of the Murine Hind Limb Vasculature and Muscle Tissue by MicroCT and Light Microscopy</strong> <div class="csl-block">Laura Schaad, Ruslan Hlushchuk, Sébastien Barré, Roberto Gianni-Barrera, David Haberthür, Andrea Banfi, Valentin Djonov</div> <em>Scientific Reports</em> (2017-02-07) <a href="https://doi.org/f9p4pn">https://doi.org/f9p4pn</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/srep41842">10.1038/srep41842</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28169309">28169309</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5294414">PMC5294414</a></div></div>
</div>
<div id="ref-cNoYNu8B" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">32. </div><div class="csl-right-inline"><strong>Morphological Aspects of Tumor Angiogenesis</strong> <div class="csl-block">Ruslan Hlushchuk, Sébastien Barré, Valentin Djonov</div> <em>Methods in Molecular Biology</em> (2016) <a href="https://doi.org/gr737s">https://doi.org/gr737s</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1007/978-1-4939-3999-2_2">10.1007/978-1-4939-3999-2_2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27858352">27858352</a></div></div>
</div>
<div id="ref-C9ratYNO" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">33. </div><div class="csl-right-inline"><strong>Therapeutic Targeting of CD146/MCAM Reduces Bone Metastasis in Prostate Cancer</strong> <div class="csl-block">Eugenio Zoni, Letizia Astrologo, Charlotte KY Ng, Salvatore Piscuoglio, Janine Melsen, Joël Grosjean, Irena Klima, Lanpeng Chen, Ewa B Snaar-Jagalska, Kenneth Flanagan, … George N Thalmann</div> <em>Molecular Cancer Research</em> (2019-05-01) <a href="https://doi.org/gr7379">https://doi.org/gr7379</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1158/1541-7786.mcr-18-1220">10.1158/1541-7786.mcr-18-1220</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30745464">30745464</a></div></div>
</div>
<div id="ref-16lhxZOpa" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">34. </div><div class="csl-right-inline"><strong>Mouse models for studying prostate cancer bone metastasis</strong> <div class="csl-block">Jinlu Dai, Janine Hensel, Ning Wang, Marianna Kruithof-de Julio, Yusuke Shiozawa</div> <em>BoneKEy Reports</em> (2016-02-17) <a href="https://doi.org/gr7372">https://doi.org/gr7372</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1038/bonekey.2016.4">10.1038/bonekey.2016.4</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26916039">26916039</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4757481">PMC4757481</a></div></div>
</div>
<div id="ref-Siy9vGW7" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">35. </div><div class="csl-right-inline"><strong>The miniature pig: a useful large animal model for dental and orofacial research</strong> <div class="csl-block">S Wang, Y Liu, D Fang, S Shi</div> <em>Oral Diseases</em> (2007-11) <a href="https://doi.org/cs8xz4">https://doi.org/cs8xz4</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/j.1601-0825.2006.01337.x">10.1111/j.1601-0825.2006.01337.x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17944668">17944668</a></div></div>
</div>
<div id="ref-i45b7Cmp" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">36. </div><div class="csl-right-inline"><strong>Animal models for implant biomaterial research in bone: A review</strong> <div class="csl-block">AI Pearce, RG Richards, S Milz, E Schneider, SG Pearce</div> <em>European Cells and Materials</em> (2007-03-02) <a href="https://doi.org/gf9rrt">https://doi.org/gf9rrt</a> <div class="csl-block">DOI: <a href="https://doi.org/10.22203/ecm.v013a01">10.22203/ecm.v013a01</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17334975">17334975</a></div></div>
</div>
<div id="ref-1EXdAADA3" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">37. </div><div class="csl-right-inline"><strong>Minipig Models of Diabetes Mellitus</strong> <div class="csl-block">Dwight Bellinger, Elizabeth Merricks, Timothy Nichols</div> <em>The Minipig in Biomedical Research</em> (2011-12-19) <a href="https://doi.org/fx5bng">https://doi.org/fx5bng</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1201/b11356-37">10.1201/b11356-37</a></div></div>
</div>
<div id="ref-lVl4gVyN" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">38. </div><div class="csl-right-inline"><strong>Tissue Engineered Neurovascularization Strategies for Craniofacial Tissue Regeneration</strong> <div class="csl-block">Yiming Li, David Fraser, Jared Mereness, Amy Van Hove, Sayantani Basu, Maureen Newman, Danielle SW Benoit</div> <em>ACS Applied Bio Materials</em> (2021-11-29) <a href="https://doi.org/gr737z">https://doi.org/gr737z</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1021/acsabm.1c00979">10.1021/acsabm.1c00979</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35014834">35014834</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9016342">PMC9016342</a></div></div>
</div>
<div id="ref-Ha1wHU4s" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">39. </div><div class="csl-right-inline"><strong>Coupling Osteogenesis and Vasculogenesis in Engineered Orthopedic Tissues</strong> <div class="csl-block">Nicholas G Schott, Nicole E Friend, Jan P Stegemann</div> <em>Tissue Engineering Part B: Reviews</em> (2021-06-01) <a href="https://doi.org/gqx6tq">https://doi.org/gqx6tq</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1089/ten.teb.2020.0132">10.1089/ten.teb.2020.0132</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32854589">32854589</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8349721">PMC8349721</a></div></div>
</div>
<div id="ref-zB5E7Obz" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">40. </div><div class="csl-right-inline"><strong>Blood vessel formation and function in bone</strong> <div class="csl-block">Kishor K Sivaraj, Ralf H Adams</div> <em>Development</em> (2016-08-01) <a href="https://doi.org/gmgfvb">https://doi.org/gmgfvb</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1242/dev.136861">10.1242/dev.136861</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27486231">27486231</a></div></div>
</div>
<div id="ref-16mB4VUpP" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">41. </div><div class="csl-right-inline"><strong>Engineering blood vessels and vascularized tissues: technology trends and potential clinical applications</strong> <div class="csl-block">Prafulla Chandra, Anthony Atala</div> <em>Clinical Science</em> (2019-05) <a href="https://doi.org/gr7374">https://doi.org/gr7374</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1042/cs20180155">10.1042/cs20180155</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/31088895">31088895</a></div></div>
</div>
<div id="ref-1ByuShsGE" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">42. </div><div class="csl-right-inline"><strong>A method to quantify and visualize femoral head intraosseous arteries by micro-CT</strong> <div class="csl-block">Xing Qiu, Xiaotian Shi, Jun Ouyang, Dachuan Xu, Dewei Zhao</div> <em>Journal of Anatomy</em> (2016-04-14) <a href="https://doi.org/f9d95v">https://doi.org/f9d95v</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1111/joa.12475">10.1111/joa.12475</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27074892">27074892</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4948056">PMC4948056</a></div></div>
</div>
<div id="ref-15Cb0RCtH" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">43. </div><div class="csl-right-inline"><strong>Impact of Anti-Angiogenic Treatment on Bone Vascularization in a Murine Model of Breast Cancer Bone Metastasis Using Synchrotron Radiation Micro-CT</strong> <div class="csl-block">Hao Xu, Marie-Hélène Lafage-Proust, Lamia Bouazza, Sandra Geraci, Philippe Clezardin, Bernard Roche, Françoise Peyrin, Max Langer</div> <em>Cancers</em> (2022-07-15) <a href="https://doi.org/gr738f">https://doi.org/gr738f</a> <div class="csl-block">DOI: <a href="https://doi.org/10.3390/cancers14143443">10.3390/cancers14143443</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/35884504">35884504</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9321934">PMC9321934</a></div></div>
</div>
<div id="ref-D7D7Wyla" class="csl-entry" role="doc-biblioentry">
<div class="csl-left-margin">44. </div><div class="csl-right-inline"><strong>Metastatic bone disease: Pathogenesis and therapeutic options</strong> <div class="csl-block">Stella D'Oronzo, Robert Coleman, Janet Brown, Francesco Silvestris</div> <em>Journal of Bone Oncology</em> (2019-04) <a href="https://doi.org/gr737x">https://doi.org/gr737x</a> <div class="csl-block">DOI: <a href="https://doi.org/10.1016/j.jbo.2018.10.004">10.1016/j.jbo.2018.10.004</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30937279">30937279</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6429006">PMC6429006</a></div></div>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p,
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* references */
  /* -------------------------------------------------- */

  .csl-entry {
    margin-top: 15px;
    margin-bottom: 15px;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen()) ||
        heading.dataset.collapsed === "true"
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery =
    'h1, h2, h3, div[id^="fig:"], div[id^="tbl:"], span[id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script on DOMContentLoaded instead of load to ensure this plugins
  // runs before other plugins
  window.addEventListener("DOMContentLoaded", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false")
      makeButtons(`div[id^="ref-"]`, ".csl-left-margin", "reference");
    if (figures !== "false")
      makeButtons(`div[id^="fig:"]`, "figcaption", "figure");
    if (tables !== "false") makeButtons(`div[id^="tbl:"]`, "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(query);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
      "class",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML =
      document.querySelector(".icon_caret_right").innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector(":nth-child(2)");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script type="module">
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
</body>
</html>
